org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/DefaultGradleBuild.java

```diff
@@ -157,11 +159,11 @@ public boolean equals(Object obj) {
      */
     private static class SynchronizeOperation extends BaseToolingApiOperation {
 
-        private final InternalGradleBuild gradleBuild;
+        private final DefaultGradleBuild gradleBuild;
         private final NewProjectHandler newProjectHandler;
         private List<SynchronizationProblem> failures;
 
-        public SynchronizeOperation(InternalGradleBuild gradleBuild, NewProjectHandler newProjectHandler) {
+        public SynchronizeOperation(DefaultGradleBuild gradleBuild, NewProjectHandler newProjectHandler) {
```

Comment:
❌ A CachingGradleBuild decorator should be used here, so it is naturally scoped to the sync operation instead of mutating state inside the GradleBuild object. The DefaultGradleBuild should have no caching logic.